CHANGELOG.md

```diff
@@ -1,3 +1,971 @@
+  - JSONP callback must be specified by `jsonpCallbackParam` config (fb663418710736161a6b5da49c345e92edf58dcb #15161 #11352)
+  - JSONP requests now require a trusted resource URL (6476af83cd0418c84e034a955b12a842794385c4 #11352)
+- **ngModelOptions:** allow options to be inherited from ancestor `ngModelOptions` (87a2ff76af5d0a9268d8eb84db5755077d27c84c #10922)
+- **input:** add support for binding to `input[type=range]` (#14870) (9130166767c4792c5d32d08a918fc7becf32c9a6 #5892)
+- **ngRoute:** allow `ngView` to be included in an asynchronously loaded template (c13c666728c1a1485ef18e92d7cb35118ce39609 #1213)
+- **select:** support values of any type added with `ngValue` (f02b707b5e4a5ffd1e1a20d910754cfabfc19622 #9842)
+- **$interpolate:** use custom `toString()` function if present (a5fd2e4c0376676fa317e09a8d8be4966b82cbfe #7317 #11406)
+- **$route:** implement `resolveRedirectTo` (e9865654b39c71be71034c38581a8c7bd16bc716 #5150)
+- **input[radio]:** allow `ng-trim` to work for `input[type=radio]` (47724baffe050269385b3481e9a9cf4ab3944b4b)
+- **$q:** report promises with non rejection callback (c9dffde1cb167660120753181cb6d01dc1d1b3d0)
+- **$location:** default hashPrefix to `&#39;!&#39;` (aa077e81129c740041438688dff2e8d20c3d7b52 #13812)
+
+
+## Performance Improvements
+- **form, ngModel:** change controllers to use prototype methods (9e24e774a558143b3478536911a3a4c1714564ba)
+- **select:** don&#39;t prepend unknown option if already prepended (ba36bde6736f0810ca670e10952a8e1c021de531)
+- **$animate:** listen for document visibility changes (d71dc2f5afec230711351e9f160873a41eb60597)
+- **injector:** cache the results of the native class detection check (5ceb5dbfa6d9b6d15232a1f5c767b2f431325948)
+- **$parse:** Inline constants (bd7d5f6345439aa2d1da708ffee20b4c565131d4)
+- **$compile:** use strict comparison for `controller === &#39;@&#39;` (bbd3db14f857aab996ad129f2f15ca6348e9fd9f)
+
+
+## Breaking Changes
+
+- **fix(jqLite): camelCase keys in jqLite#data (fc0c11db845d53061430b7f05e773dcb3fb5b860)**:
+
+Previously, keys passed to the data method were left untouched.
+Now they are internally camelCased similarly to how jQuery handles it, i.e.
+only single (!) hyphens followed by a lowercase letter get converted to an
+uppercase letter. This means keys `a-b` and `aB` represent the same data piece;
+writing to one of them will also be reflected if you ask for the other one.
+
+If you use Angular with jQuery, it already behaved in this way so no changes
+are required on your part.
+
+To migrate the code follow the examples below:
+
+BEFORE:
+
+/* 1 */
+elem.data(&#39;my-key&#39;, 2);
+elem.data(&#39;myKey&#39;, 3);
+
+/* 2 */
+elem.data(&#39;foo-bar&#39;, 42);
+elem.data()[&#39;foo-bar&#39;]; // 42
+elem.data()[&#39;fooBar&#39;]; // undefined
+
+/* 3 */
+elem.data()[&#39;foo-bar&#39;] = 1;
+elem.data()[&#39;fooBar&#39;] = 2;
+elem.data()[&#39;foo-bar&#39;]; // 1
```

Comment:
These sections need to be wrapped in triple backticks, because all lines render on the same line.
(Also `&#39;` --> `'` :grin:)